soh/CMake/Linux32bit-toolchain.cmake

```diff
@@ -0,0 +1,18 @@
+# the name of the target operating system
```

Comment:
Now that macOS is merged in, we'll probably want to have a `Mac-toolchain.cmake`.

Reply:
It would likely look like this:
```
# the name of the target operating system
set(CMAKE_SYSTEM_NAME Darwin)

# which compilers to use for C and C++
set(CMAKE_C_COMPILER gcc)
set(CMAKE_CXX_COMPILER g++)
set(CMAKE_CXX_FLAGS -fpermissive)

# adjust the default behaviour of the FIND_XXX() commands:
# search headers and libraries in the target environment, search
# programs in the host environment
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
```

Reply:
sure, mac-toolchain can be added easily. Would you need mac-gnu-toolchain and mac-clang-toolchain?

Reply:
Eventually we'll also need a Mac-clang -- but currently the project doesn't build with clang. We have a separate effort to make the code compilable under clang.

But if it did work we'd have to update all:

```cmake
if(NOT CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
# to
if(NOT CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang|AppleClang")
```